packages/lit-html/src/lib/directives/cache.ts

```diff
@@ -0,0 +1,73 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {
+  directive,
+  TemplateResult,
+  NodePart,
+  NodePartState,
+  Directive,
+} from '../lit-html.js';
+
+/**
+ * Enables fast switching between multiple templates by caching the DOM nodes
+ * and TemplateInstances produced by the templates.
+ *
+ * Example:
+ *
+ * ```
+ * let checked = false;
+ *
+ * html`
+ *   ${cache(checked ? html`input is checked` : html`input is not checked`)}
+ * `
+ * ```
+ */
+export const cache = directive(
+  class extends Directive {
+    templateCache = new WeakMap<TemplateStringsArray, NodePartState>();
+    value?: TemplateResult;
+
+    render(v: unknown) {
+      if ((v as TemplateResult)._$litType$ !== undefined) {
```

Comment:
`v` can be nullish here, right?
```suggestion
      if ((v as TemplateResult | null)?._$litType$ !== undefined) {
```